lib/avromatic/model/raw_serialization.rb

```diff
@@ -29,7 +29,7 @@ def recursive_serialize(value, name: nil, member_types: nil)
             if value.is_a?(Avromatic::Model::Attributes)
               hash = value.value_attributes_for_avro
               if Avromatic.use_custom_datum_writer
-                unless value.class.config.mutable
+                if Avromatic.use_encoding_providers && !value.class.config.mutable
```

Comment:
Minor but often times Ruby code has both positive a negative forms of boolean methods e.g. `mutable` and `immutable?` for a slight improvement in readability. Also this lines relates to my earlier question about the relation between `use_custom_datum_writer` and `use_encoding_providers`.

Reply:
I like this idea in general but in this particular case it strikes me as potentially confusing. My concern is that someone might read value.class.config.immutable? and interpret it as "is the value.class.config object immutable?".

Reply:
Fair point

Reply:
It might make sense to define `mutable?/immutable?` as static methods on the generated classes that delegate to the config. I've come close to doing that a couple of times. And going one step further instance could delegate to a method on the class.